main.go

```diff
@@ -130,13 +133,15 @@ func main() {
 	// and we want to loosen some assumptions, eg. doesn't matter if not paired.
 	vehicleInfo, err := blockingGetVehicleInfo(logger, ethAddr, lss, env)
 	if err != nil {
-		logger.Fatal().Err(err).Msgf("cannot start edge-network because no on-chain pairing was found for this device addr")
+		logger.Fatal().Err(err).Msgf("cannot start edge-network because no on-chain pairing was found for this device addr: %s", ethAddr.Hex())
 	}
 
 	// OBD / CAN Loggers
-	ds := network.NewDataSender(unitID, *ethAddr, logger, vehicleInfo)
+	// we still need to set vehicleInfo, I know this is ugly, but..
+	ds.SetVehicleInfo(vehicleInfo)
 	if ethErr != nil {
-		logger.Info().Msgf("error getting ethereum address: %s", err)
+		// would we ever end up here?
+		logger.Err(ethErr).Msgf("error getting ethereum address: %s", err)
```

Comment:
it could happen, and would be a fatal failure, but current datasender payload I think does not include serial (unitID) which would be the only other way to identify the device.

Reply:
oh, I see we already have a check for this further up, this is probably legacy should be deleted starting on line 142